h/static/scripts/tests/controllers/share-widget-controller-test.js

```diff
@@ -0,0 +1,144 @@
+          {{ svg_icon('google-plus', 'share-widget-action') }}
+        </a>
+        <a share-target-href="mailto:?subject=Let's%20Annotate&amp;body={href}"
+          class="share-widget-target__icon">
+        </a>
+      </div>
+      <div class="share-widget-link">
+          <input class="js-share-widget-clipboard" value="">
+      </div>
+      <div class="js-share-widget-msg-group">GROUP</div>
+      <div class="js-share-widget-msg-private">PRIVATE</div>
+    </div>
+  </div>
+  `;
+
+  const template = `
+    <button share-widget-config='{"url":"url1", "private":false, "group":false}'>
+      btn
+    </button>
+    <button share-widget-config='{"url":"url2", "private":true, "group":false}'>
+      btn
+    </button>
+    <button share-widget-config='{"url":"url3", "private":false, "group":true}'>
+      btn
+    </button>
+    <button share-widget-config='{"url":"url4", "private":true, "group":true}'>
+      <span><b>btn</b></span>
+    </button>
+  `;
+
+  let container;
+  let ctrl;
+
+  const getTriggers = () => {
+    return Array.from(container.querySelectorAll('[share-widget-config]'));
+  };
+
+  const widgetIsVisble = () => {
+    return container.querySelector('.js-share-widget').style.visibility === 'visible';
+  };
+
+  const urlsReflectUpdate = (interpolatedValue) => {
+    return Array.from(container.querySelectorAll('[share-target-href]')).every((anchor) => {
+      const tmpl = anchor.getAttribute('share-target-href');
+      return tmpl.replace('{href}', interpolatedValue) === anchor.href;
+    });
+  };
+
+  const privateMessageVisible = () => {
+    return container.querySelector('.js-share-widget-msg-private').style.display === 'block';
+  };
+
+  const groupMessageVisible = () => {
+    return container.querySelector('.js-share-widget-msg-group').style.display === 'block';
+  };
+
+  beforeEach(() => {
+    container = document.createElement('div');
+    container.innerHTML = template + widgetTemplate;
+
+    document.body.appendChild(container);
+
+    ctrl = new ShareWidgetController(document.querySelector('.js-share-widget'));
+
+  });
+
+  afterEach(() => {
+    container.remove();
+    ctrl.beforeRemove();
+  });
+
+  it('shows/hides widget on clicking', (done) => {
+
+    assert.isFalse(widgetIsVisble(), 'not visible by default');
+
+    const btns = getTriggers();
+
+    syn
+      .click(btns[0], () => {
+        assert.isTrue(widgetIsVisble(), 'visible after clicking trigger');
+      })
+      .click(btns[1], () => {
+        assert.isFalse(widgetIsVisble(), 'hidden after clicking another trigger while it is open');
+      })
+      .click(btns[3].querySelector('b'), () => {
+        assert.isTrue(widgetIsVisble(), 'opens when clicking elements inside of trigger');
+        done();
+      });
+  });
+
+  it('displays correctly for current share link', (done) => {
+
+    const btns = getTriggers();
+
+    syn
+      .click(btns[0], () => {
+        assert.isTrue(urlsReflectUpdate('url1'));
```

Comment:
I think it makes tests more useful and easier to understand if the test data (ie. the URL in this case) follows the same format as actual data that might be used. In other words, this should be a real URL - since the test needs to check that URL encoding works correctly.